### PR TITLE
Solution implemented

### PR DESCRIPTION
**Pull request by NikolayAbysov**

### CODE REVIEW

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                    + "get product query. Stack trace: "
+                    + e.getMessage());
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        List<Order> orders = new ArrayList<>();
+        String query = "SELECT * "
+                + "FROM internetshop.orders;";
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java]:
```suggestion
                + "FROM orders;";
```
you setting database name when setting connection

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        Long orderId = resultSet.getLong("order_id");
+        Long userId = resultSet.getLong("user_id");
+        return new Order(orderId, getProducts(orderId), userId);
+    }
+
+    private List<Product> getProducts(Long orderId) {
+        String query = "SELECT products.product_id, products.name, products.price "
+                + "FROM orders_products "
+                + "INNER JOIN products "
+                + "ON  orders_products.product_id=products.product_id "
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java]:
```suggestion
                + "USING(product_id) "
```

File: src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        try {
+            deleteProducts(shoppingCart);
+        } catch (DataProcessingException e) {
+            addProducts(shoppingCart);
+            return shoppingCart;
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java]: do we need this try catch?

File: src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    private void addRoles(User user) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Role role: user.getRoles()) {
+                String query = "INSERT INTO internetshop.users_roles (user_id, role_id) "
+                        + "VALUES (?,?)";
+                PreparedStatement statement = connection.prepareStatement(query);
+                statement.setLong(1, user.getId());
+                statement.setLong(2, role.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoImpl.java]: where did you get roleId at this stage?

File: src/main/java/mate/academy/internetshop/model/Role.java [hunk 5]
```diff
@@ -6,6 +6,11 @@
 
     public Role(RoleName roleName) {
         this.roleName = roleName;
+        if (roleName.equals(RoleName.ADMIN)) {
+            this.id = 1L;
+        } else {
+            this.id = 2L;
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/model/Role.java]: that's were)) imagine we will have 10 roles or even more?

**Review by dmytro-hryhoruk — COMMENTED** [review]
Remove database name from all queries

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java (hunk 1 above)
Comment (NikolayAbysov) [reply src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java]: Hi! Yep, fixed

File: src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java (hunk 3 above)
Comment (NikolayAbysov) [reply src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java]: At some moment it was REALLY needed :)

File: src/main/java/mate/academy/internetshop/model/Role.java (hunk 5 above)
Comment (NikolayAbysov) [reply src/main/java/mate/academy/internetshop/model/Role.java]: Should i implement role service?
It seems the solution, but big one...

Comment (NikolayAbysov) [reply src/main/java/mate/academy/internetshop/model/Role.java]: Decided to use name of Role to get its Id. Maybe, not brilliant, but alternative is to implement role service :)

File: src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+                    + e.getMessage());
+        }
+    }
+
+    private void addProducts(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : order.getProducts()) {
+                String query = "INSERT INTO orders_products(order_id, product_id) "
+                        + "values(?,?)";
+                PreparedStatement statement = connection.prepareStatement(query);
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoImpl.java]: let's move creating statement from loop

File: src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                    + e.getMessage());
+        }
+    }
+
+    private void addProducts(ShoppingCart shoppingCart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Product product : shoppingCart.getProducts()) {
+                String query = "INSERT INTO "
+                        + "shopping_cart_products(cart_id, product_id) "
+                        + "values(?,?);";
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/ShoppingCartDaoImpl.java]: same

File: src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+        }
+        return roles;
+    }
+
+    private void setRolesId(Set<Role> roles) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            for (Role role : roles) {
+                String query = "SELECT role_id FROM roles WHERE role_name = ?";
+                PreparedStatement preparedStatement = connection.prepareStatement(query);
+                preparedStatement.setString(1, role.getRoleName().toString());//check!
```
Comment (dmytro-hryhoruk) [inline src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoImpl.java]: )

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**